internal/linux/memfd.go

```diff
@@ -0,0 +1,39 @@
+package linux
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+// CreateMemfd creates a new memfd for the provided byte slice.
+func CreateMemfd(content []byte) (*os.File, error) {
+	// Create the memfd.
+	fd, err := unix.MemfdCreate("memfd", unix.MFD_CLOEXEC)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set its size.
+	err = unix.Ftruncate(fd, int64(len(content)))
+	if err != nil {
```

Comment:
Hm, in tihs case and any case where you return before returning os.Newfile(fd), should you be closing fd?

Reply:
Yep, I definitely should :)

Reply:
Revert logic added which will now close the fd if anything goes wrong in the function.